packages/client/hooks/useTransition.ts

```diff
@@ -73,14 +73,17 @@ const useTransition = <T extends {key: Key}>(children: T[]) => {
 
   return useMemo(() => {
     const currentTChildren = [] as TransitionChild<T>[]
-    const {current: prevTChildren} = previousTransitionChildrenRef
+    const filteredPrevTChildren = previousTransitionChildrenRef.current.filter(
+      (prevTChild) => prevTChild.status !== TransitionStatus.EXITING
+    )
```

Comment:
@mattkrick do you mind taking a look at the changes in this file? we had a case where elements were not showing after back & forth navigation, more details here: https://github.com/ParabolInc/parabol/pull/7542#pullrequestreview-1210944164 

thank you!

Reply:
i can test this later today after the ship, but just from reading the code, it looks like animations that are exiting will get set to mounted, which means the exit animations will cut short. if that's not the case, LGTM!

i know framer-motion does this really well under the hood, if it's easier to use that, I wouldn't be opposed

Reply:
@mattkrick The case you mentioned would specifically happen for children that are currently present. I.e. a child that _was_ exiting as a result of no longer being present, but is now once again present, would be treated as though there’s no previous child state.

The specific issue I found while debugging is as follows:
* Set X of children are mounted
* Transition to new meeting occurs, so set X transitions to exiting, and set Y transitions through entering, entered, mounted. Set X is still kept in prev children with state exiting.
* Transition to the previous meeting occurs, so set Y transitions to exiting, and set X _should_ transition back to mounted, but because they already exist on prevChildren, we keep their state, which is exiting.

Normally, the exiting child would be removed from the prev children when its transition ends, but there seems to be a react-y DOM reuse thing happening so the `onTransitionEnd` event doesn't fire, because the would-be exiting cards in the DOM are reused to render the new meeting's cards.

The solution implemented here is treating children that were in the exiting state previously as "out of the picture".

Reply:
> set X should transition back to mounted, but because they already exist on prevChildren, we keep their state, which is exiting.

ahh, that's the part i needed to read in prose. Got it, LGTM!